r-package/policytree/src/tree_search.cpp

```diff
@@ -374,5 +423,66 @@ std::unique_ptr<Node> tree_search(int depth, int split_step, size_t min_node_siz
     v.resize(num_points + 1, 0.0);
   }
 
-  return find_best_split(sorted_sets, depth, split_step, min_node_size, data, sum_array);
+  return find_best_split(sorted_sets, depth, split_step, min_node_size, data, sum_array, 0);
+}
+
+std::unique_ptr<Node> tree_search_hybrid(int max_global_depth, int complete_split_depth, int chop_depth, int repeat_splits, int split_step, size_t min_node_size, const Data* data) {
+  size_t num_rewards = data->num_rewards();
+  size_t num_points = data->num_rows;
+  auto sorted_sets = create_sorted_sets(data);
+
+  std::vector<std::vector<double>> sum_array;
+  sum_array.resize(num_rewards);
+  for (auto& v : sum_array) {
+    // + 1 because this is a cumulative sum of rewards, entry 0 will be zero.
+    v.resize(num_points + 1, 0.0);
+  }
+
+  auto start = new Node(0, 0.0, 0, 0, 0, 0);
+      //TODO(kanodiaayush):uncomment
+  start->complete_sorted_sets = sorted_sets;
+  std::queue<Node*> expansion_queue;
+  expansion_queue.push(start);
+  while(!expansion_queue.empty()) {
+    auto expansion_node = expansion_queue.front();
+    expansion_queue.pop();
+    if(expansion_node->height < 1) {
+	    continue;
```

Comment:
this first node is initialized with height=0 so this loop will terminate immediately ... ?